nondex-common/src/main/java/edu/illinois/nondex/common/SystematicRandom.java

```diff
@@ -0,0 +1,144 @@
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Random;
+import java.util.Stack;
+import java.util.logging.Level;
+
+public class SystematicRandom extends Random {
+    private final Stack<ExplorationEntry> choices;
+    private final String logFileName;
+    private final Configuration config = Configuration.parseArgs();
+    private List<String> lines;
+    private int replayIndex;
+
+    public SystematicRandom() {
+        logFileName = config.systematicLog;
+        choices = new Stack<ExplorationEntry>();
+        File file = new File(logFileName);
+        if (file.exists()) {
+            try {
+                lines = Files.readAllLines(Paths.get(logFileName));
+            } catch (IOException ioe) {
+                Logger.getGlobal().log(Level.SEVERE,"Could not read lines from systematic.log" ,ioe);
+            }
+            Object[] choiceValues;
+            for (String element: lines) {
+                String delimiter = "[ ]+";
+                if (!element.isEmpty()) {
+                    choiceValues = element.split(delimiter);
+                    if (choiceValues.length == 3) {
+                        int current = Integer.parseInt(choiceValues[0].toString());
+                        int maximum = Integer.parseInt(choiceValues[1].toString());
+                        boolean shouldExplore = Boolean.parseBoolean(choiceValues[2].toString());
+                        ExplorationEntry currentMaximum = new ExplorationEntry(current, maximum, shouldExplore);
+                        choices.push(currentMaximum);
+                    } else {
+                        Logger.getGlobal().log(Level.SEVERE, "The 3 ExplorationEntry variable were not stored properly");
+                    }
+
+                }
+            }
+        }
+    }
+
+    public int nextInt(final int maximum) {
+        int current;
+        boolean explore;
+        if (replayIndex < choices.size()) {
+            current =  choices.get(replayIndex).getCurrent();
+        } else {
+            current = 0;
+            if (choices.size() > config.start) {
+                explore = true;
+            } else {
+                explore = false;
+            }
+            ExplorationEntry choiceNums = new ExplorationEntry(current, maximum, explore);
+            choices.push(choiceNums);
+        }
+        replayIndex++;
+        return current;
+    }
+
+    public void endRun() {
+        while (!choices.isEmpty()) {
+            ExplorationEntry currentMaximum = choices.pop();
+            int current = currentMaximum.getCurrent();
+            int maximum = currentMaximum.getMaximum();
+            boolean shouldExplore = false;
+            if (current < maximum - 1) {
+                current++;
+                if (choices.size() > config.start) {
+                    shouldExplore = true;
+                }
+                currentMaximum.setCurrent(current);
+                currentMaximum.setShouldExplore(shouldExplore);
```

Comment:
Something is fishy with the shouldExplore field of the entry. You are only writing it, and never using it in your logic that returns a random number. I think you should use it there, or if you think you shouldn't please remove it from everywhere.